packages/opentelemetry-core/src/context/propagation/B3Format.ts

```diff
@@ -42,29 +43,29 @@ function isValidSpanId(spanId: string): boolean {
  * Based on: https://github.com/openzipkin/b3-propagation
  */
 export class B3Format implements HttpTextFormat {
-  inject(context: Context, carrier: Carrier) {
+  inject(context: Context, carrier: unknown, setter: SetterFunction) {
     const spanContext = getParentSpanContext(context);
     if (!spanContext) return;
 
     if (
       isValidTraceId(spanContext.traceId) &&
       isValidSpanId(spanContext.spanId)
     ) {
-      carrier[X_B3_TRACE_ID] = spanContext.traceId;
-      carrier[X_B3_SPAN_ID] = spanContext.spanId;
+      setter(carrier, X_B3_TRACE_ID, spanContext.traceId);
+      setter(carrier, X_B3_SPAN_ID, spanContext.spanId);
 
       // We set the header only if there is an existing sampling decision.
       // Otherwise we will omit it => Absent.
       if (spanContext.traceFlags !== undefined) {
-        carrier[X_B3_SAMPLED] = Number(spanContext.traceFlags);
+        setter(carrier, X_B3_SAMPLED, Number(spanContext.traceFlags));
       }
     }
   }
 
-  extract(context: Context, carrier: Carrier): Context {
-    const traceIdHeader = carrier[X_B3_TRACE_ID];
-    const spanIdHeader = carrier[X_B3_SPAN_ID];
-    const sampledHeader = carrier[X_B3_SAMPLED];
+  extract(context: Context, carrier: unknown, getter: GetterFunction): Context {
```

Comment:
should the `getter` be optional and if missed the `defaultGetter` can be used - this would simplify things in most cases. - the same would apply for `setter` in `inject`

Reply:
This is only called by the global API, which does just that. Users should not be calling this directly.